pkg/controller/certificate-shim/sync.go

```diff
@@ -231,23 +231,26 @@ func validateGatewayListenerBlock(path *field.Path, l gwapi.Listener) field.Erro
 		return errs
 	}
 
-	if l.TLS.CertificateRef == nil {
+	if len(l.TLS.CertificateRefs) == 0 {
 		errs = append(errs, field.Required(path.Child("tls").Child("certificateRef"),
-			"listener is missing a certificateRef"))
+			"listener has no certificateRefs"))
 	} else {
-		if l.TLS.CertificateRef.Group != "core" {
-			errs = append(errs, field.NotSupported(path.Child("tls").Child("certificateRef").Child("group"),
-				l.TLS.CertificateRef.Group, []string{"core"}))
-		}
+		// check that each CertificateRef is valid
+		for i, secretRef := range l.TLS.CertificateRefs {
+			if secretRef == nil {
+				errs = append(errs, field.Required(path.Child("tls").Child("certificateRef").Index(i), "listener is missing a certificateRef"))
+				continue
+			}
 
-		if l.TLS.CertificateRef.Kind != "Secret" {
-			errs = append(errs, field.NotSupported(path.Child("tls").Child("certificateRef").Child("kind"),
-				l.TLS.CertificateRef.Kind, []string{"Secret"}))
-		}
+			if *secretRef.Group != "core" && *secretRef.Group != "" {
+				errs = append(errs, field.NotSupported(path.Child("tls").Child("certificateRef").Index(i).Child("group"),
+					*secretRef.Group, []string{"core", ""}))
+			}
```

Comment:
we should document the fact that the only two `certificateRefs` supported by cert-manager are `group: core` or with the empty `group`:

```yaml
certificateRefs:
  - kind: Secret
    group: core
    name: secret-1
    namespace: default
```

or


```yaml
certificateRefs:
  - kind: Secret
    name: secret-1
    namespace: default
```

and that `namespace` may be omitted, in which case the HTTPRoute's namespace is used.

Reply:
👍 Yes, I will update our website before 1.8.